cfpb/cfgov-refresh
# Moving OAH into cfgov-refresh

## Removed
Removed anything related to `loan-comparison` and react. This was all dead code which was deemed unnecessary.

## Additions
- Added all the owning a home assets into their own folder. This was done for jinja templates, css, js, and images. The static file asset urls were updated to reflect their new location.

## Changes

-  Modified `urls.py` so I wouldn't have to create 9000 flagged routes. I ended up creating `cfgov/v1/views/oah.py` which handles all the owning a home requests. Not sure it's a great idea but I did it anyway. I have regrets.

- Modified `cfgov/unprocessed/js/organisms/Expandable.js` to fix issues with the accordion and race conditions. Will mirror the changes in a separate PR. 

- Modified the front-end build process to accommodate owning a home.

- Modified the JS unit test to work with our existing architecture.


## Testing

Copy the `cfgov/unprocessed/js/owning-a-home/config/example-config.json` file to 
`cfgov/unprocessed/js/owning-a-home/config/config.json` as listed here (step 4): https://github.com/cfpb/owning-a-home#node-grunt-bower-browserify

1. Run `gulp clean`.
2. Run `python cfgov/manage.py collectstatic --clear`.
3. Run `setup.sh` and follow the instructions to setup OAH and OAH-API.
4. Run `gulp images` ( there is some strange bug so this is necessary ).
5. Goto Wagtail and enable `OAH_DASHEERING` flag by setting it to the following:
<img width="1135" alt="screen shot 2017-07-23 at 5 26 22 am" src="https://user-images.githubusercontent.com/1696212/28498274-7e37f1ac-6f67-11e7-89aa-71587aa49e2b.png">

6. Visit the following urls and verify that they work.
7. Go back to Wagtail and delete the  `OAH_DASHEERING` flag.
8. Visit the following urls and verify that they work.
9. Goto Wagtail and visit all `owning-a-home` pages and verify that they work.
10. Run `gulp test:unit` and verify that the test pass.
11. Visit the same URLS in production and perform a visual comparison.
12. If you have continued reading to this point, it's impressive. You've reached level 5. Your reward is 
     more testing.
13. Visit the production equivalents of the urls listed below and perform a visual comparison.

http://localhost:8000/owning-a-home/

http://localhost:8000/static/owning-a-home/resources/adjustable_rate_loan_estimate.pdf
http://localhost:8000/static/owning-a-home/resources/checklist_mortgage_closing.pdf
http://localhost:8000/static/owning-a-home/resources/monthly_payment_worksheet.pdf
http://localhost:8000/static/owning-a-home/resources/mortgage_closing_forms.pdf

http://localhost:8000/owning-a-home/closing-disclosure/
http://localhost:8000/owning-a-home/explore-rates/
http://localhost:8000/owning-a-home/loan-estimate/

http://localhost:8000/owning-a-home/loan-options/
http://localhost:8000/owning-a-home/loan-options/conventional-loans/
http://localhost:8000/owning-a-home/loan-options/special-loan-programs/
http://localhost:8000/owning-a-home/loan-options/FHA-loans/

http://localhost:8000/owning-a-home/mortgage-closing/
http://localhost:8000/owning-a-home/mortgage-estimate/

http://localhost:8000/owning-a-home/process/
http://localhost:8000/owning-a-home/process/prepare/
http://localhost:8000/owning-a-home/process/explore/
http://localhost:8000/owning-a-home/process/compare/
http://localhost:8000/owning-a-home/process/close/
http://localhost:8000/owning-a-home/process/sources/


## Screenshots


## Notes

-

## Todos

- Get the existing OAH acceptance tests to run. The tests start correctly but never finish.  It's like they are frozen in the tundra of Hoth. 

## Checklist
